enhancements/assets-generation/cloud-foundry-application-discovery/README.md

```diff
@@ -74,40 +74,233 @@ to implement the design.  For instance,
+| **docker** | *object* | If present, the created app will have *Docker lifecycle type*[^1]; the value of this key is ignored by the API but may be used by clients to source the registry address of the image and credentials, if needed; the [generate manifest endpoint](https://v3-apidocs.cloudfoundry.org/version/3.163.0/#generate-a-manifest-for-an-app) will return the registry address of the image and username provided with this key |
+| **env** | *object* | A key-value mapping of environment variables to be used for the app when running |
+| **processes** | *array of [process configurations](#process-level-configuration)* | List of configurations for individual process types |
+| **random-route** | *boolean* | Creates a random route for the app if `true`; if `routes` is specified, if the app already has routes, or if `no-route` is specified, this field is ignored regardless of its value |
+| **default-route** | *boolean* | If true, a route for the app will be created using the app name as the hostname and the containing organization’s default domain as the domain; if `routes` is specified, if the app already has routes, or if `no-route` is specified, this field is ignored regardless of its value |
+| **no-route** | *bool* | If false, no route is created for this application, regardless of the configuration. Note that health checks will be impacted since CF [is not able to reach](https://lists.cloudfoundry.org/g/cf-dev/topic/app_attribute_no_route_true/6333713) to the app externally to check the heart beat. This will need to be addressed in the manifest template provided by the user. |
+| **routes** | *array of [route configurations](#route-level-configuration)* | List declaring HTTP and TCP routes to be mapped to the app. |
+| **services** | *array of [service configurations](#service-level-configuration)* | A list of service-instances to bind to the app |
+| **sidecars** | *array of [sidecar configurations](#sidecar-level-configuration)* | A list of configurations for individual sidecars |
+| **stack** | *string* | The root filesystem to use with the buildpack, for example `cflinuxfs4` |
+| **metadata.labels** | [*label object*](https://v3-apidocs.cloudfoundry.org/version/3.163.0/#labels) | Labels applied to the app |
+| **metadata.annotations** | [*annotation object*](https://v3-apidocs.cloudfoundry.org/version/3.163.0/#annotations) | Annotations applied to the app |
+| **buildpack** | *string* | **DEPRECATED in favor of the `buildpacks` field above** |
+| **timeout** | *integer* | **Maximum time it can take an application to startup before CF considers it as failed. Measured in seconds** |
+
+#### Process-level configuration
+
+This configuration is for the individual process. Each process is created if it
+does not already exist. For backwards compatibility, the web process
+configuration may be placed at the top level of the application configuration,
```

Comment:
If I understand correctly, does this mean that the CF application configuration also supports process related fields if they belong to a web type? This is messy to say the least :/

Reply:
Not sure if we're thinking the same thing.
I understood that each process type can have a different configuration (e.g., different `disk_quota` or `instances`).

If there is a configuration for a process with type `web`, this "local" configuration overrides the setting specified in the application configuration.

Reply:
My concern is this statement:
```
For backwards compatibility, the web process
configuration may be placed at the top level of the application configuration.
```

Maybe I'm reading it out of context, but it looks like we could spot a process spec fields at the application spec level as well as at the sub field `Processes`.

Reply:
ok now i understand your concern and, yes it seems like that